File: Java/Medium/TimeLimitExceeded/max-sum-of-a-pair-with-equal-sum-of-digits_2.java
class Solution {
    public int maximumSum(int[] nums) {
        
        int max = -1;
        for(int i = 0; i < nums.length; i++) {
            int sum1 = sumOfDigits(nums[i]);
            for(int j = i + 1; j < nums.length; j++) {
                int sum2 = sumOfDigits(nums[j]);
                if(sum1 == sum2) {
                    max = Math.max(max, nums[i] + nums[j]);
                }
            }
        }
        return max;
    }
    
    public int sumOfDigits(int num) {
        int sum = 0;
        while(num > 0) {
            sum += num % 10;
            num /= 10;
        }
        return sum;
    }
}

File: Rust/Medium/TimeLimitExceeded/max-sum-of-a-pair-with-equal-sum-of-digits_2.rs
impl Solution {
    pub fn maximum_sum(nums: Vec<i32>) -> i32 {
        
        let mut max = -1;
        for i in 0..nums.len() {
            for j in i+1..nums.len() {
                let mut sum1 = 0;
                let mut sum2 = 0;
                let mut t = nums[i];
                while t > 0 {
                    sum1 += t % 10;
                    t /= 10;
                }
                t = nums[j];
                while t > 0 {
                    sum2 += t % 10;
                    t /= 10;
                }
                if sum1 == sum2 {
                    max = max.max(nums[i] + nums[j]);
                }
            }
        }
        max
    }
}